File: Java/09_inverter_string.java
import java.util.Scanner;

class inverter_string {

    // Forma 1: laço manual
    static String inverterManual(String texto) {
        String resultado = "";
        for (int i = texto.length() - 1; i >= 0; i--) {
            resultado += texto.charAt(i);
        }
        return resultado;
    }

    // Forma 2: array de chars
    static String inverterArray(String texto) {
        char[] chars = texto.toCharArray();
        int inicio = 0;
        int fim = chars.length - 1;
        while (inicio < fim) {
            char temp = chars[inicio];
            chars[inicio] = chars[fim];
            chars[fim] = temp;
            inicio++;
            fim--;
        }
        return new String(chars);
    }

    // Forma 3: StringBuilder
    static String inverterBuilder(String texto) {
        return new StringBuilder(texto).reverse().toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite um texto: ");
        String texto = scanner.nextLine();

        System.out.println("\nOriginal:         " + texto);
        System.out.println("Invertido (for):  " + inverterManual(texto));
        System.out.println("Invertido (array):" + inverterArray(texto));
        System.out.println("Invertido (build):" + inverterBuilder(texto));

        scanner.close();
    }
}

File: Cs/09_inverter_string.cs
using System.Text;

public class InverterString
{
    // Forma 1: laço manual
    static string InverterManual(string texto)
    {
        string resultado = "";
        for (int i = texto.Length - 1; i >= 0; i--)
        {
            resultado += texto[i];
        }
        return resultado;
    }

    // Forma 2: array de chars + Array.Reverse
    static string InverterArray(string texto)
    {
        char[] chars = texto.ToCharArray();
        Array.Reverse(chars);
        return new string(chars);
    }

    // Forma 3: StringBuilder
    static string InverterBuilder(string texto)
    {
        var sb = new StringBuilder();
        for (int i = texto.Length - 1; i >= 0; i--)
        {
            sb.Append(texto[i]);
        }
        return sb.ToString();
    }

    public static void Main(string[] args)
    {
        Console.Write("Digite um texto: ");
        string texto = Console.ReadLine();

        Console.WriteLine($"\nOriginal:            {texto}");
        Console.WriteLine($"Invertido (for):     {InverterManual(texto)}");
        Console.WriteLine($"Invertido (array):   {InverterArray(texto)}");
        Console.WriteLine($"Invertido (builder): {InverterBuilder(texto)}");
    }
}